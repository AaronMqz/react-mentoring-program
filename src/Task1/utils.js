export const GenresList = ["ALL", "DOCUMENTARY", "COMDEY", "HORROR", "CRIME"];
