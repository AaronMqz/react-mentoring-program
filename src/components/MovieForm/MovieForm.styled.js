import styled from "styled-components";

const Content = styled.div`
  display: flex;
  flex-direction: column;
  background: rgba(0, 0, 0, 0.8);
  padding: 15px;
`;

const Row = styled.div`
  display: flex;
  margin-bottom: 25px;
`;

const RowItem = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  &:nth-child(2) {
    width: 400px;
    margin-left: 20px;
  }
  position: relative;
`;

const Footer = styled.div`
  display: flex;
  justify-content: flex-end;
`;

const Title = styled.label`
  font-size: 40px;
  color: white;
  margin-bottom: 20px;
`;

const Input = styled.input`
  border: 0;
  border-radius: 4px;
  height: 40px;
  padding: 2px 10px;
  font-size: 16px;
  background: #424242;
  color: #ffffff;
`;

const Label = styled.label`
  font-size: 16px;
  color: #f64261;
  margin-bottom: 10px;
`;

const Message = styled(Label)`
  color: white;
  margin-bottom: 30px;
`;

const TextArea = styled.textarea`
  border: 0;
  background: #424242;
  border-radius: 4px;
  height: 200px;
  padding: 2px 10px;
  width: 100%;
  color: #ffffff;
  font-size: 16px;
  box-sizing: border-box;
  resize: none;
`;

const Icon = styled.span`
  display: flex;
  position: absolute;
  right: 10px;
  align-items: center;
  right: 13px;
  bottom: 15px;
`;

const ButtonPrimary = styled.div`
  border-radius: 2px;
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;
  background: #f64261;
  color: white;
  font-size: 16px;
  padding: 10px;
  margin-left: 10px;
  width: 150px;
`;

const ButtonSecondary = styled(ButtonPrimary)`
  background: #232323;
  border: solid 1px #f64261;
  color: #f64261;
`;

export default {
  Content,
  Row,
  RowItem,
  Footer,
  Title,
  Input,
  Label,
  TextArea,
  Message,
  Icon,
  ButtonPrimary,
  ButtonSecondary,
};
