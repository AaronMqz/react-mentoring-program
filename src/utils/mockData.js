export const mockData = {
  image_url: "https://m.media-amazon.com/images/I/71eHZFw+GlL._AC_SY879_.jpg",
  movie_name: "Avengers: Inifinity War",
  release_year: "2018",
  genres: ["Action", "Adventure"],
  duration: "2h 29min",
  description:
    "As the Avengers and their allies have continued to protect the world from threats too large for any one hero to handle, a new danger has emerged from the cosmic shadows: Thanos. A despot of intergalactic infamy, his goal is to collect all six Infinity Stones, artifacts of unimaginable power, and use them to inflict his twisted will on all of reality. Everything the Avengers have fought for has led up to this moment, the fate of Earth and existence has never been more uncertain.",
  rating: "8.4",
};

export const SortByOptions = [
  { id: "release_date", name: "RELEASE DATE" },
  { id: "title", name: "TITLE" },
];
