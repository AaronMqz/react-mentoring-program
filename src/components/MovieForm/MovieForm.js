import React, { useState } from "react";
import Form from "./MovieForm.styled";
import DropDown from "../DropDown/DropDown";
import { GenresList } from "../../utils";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCalendarDays } from "@fortawesome/free-solid-svg-icons";

const Default = ({ titleForm, movie, handleSubmit, handleReset }) => {
  const [newMovieState, setNewMovieState] = useState(movie);

  const {
    movie_name,
    release_date,
    image_url,
    rating,
    genres,
    duration,
    description,
  } = newMovieState;

  const handleChange = (e) => {
    setNewMovieState({
      ...newMovieState,
      [e.target.name]: e.target.value,
    });
  };

  const handleChangeGenres = (items) => {
    setNewMovieState({
      ...newMovieState,
      genres: items,
    });
  };

  return (
    <Form.Content>
      <Form.Title>{titleForm}</Form.Title>
      <Form.Row>
        <Form.RowItem>
          <Form.Label>TITLE</Form.Label>
          <Form.Input
            name={"title"}
            placeholder={"Title"}
            onChange={handleChange}
            value={movie_name}
            type="text"
          />
        </Form.RowItem>
        <Form.RowItem>
          <Form.Label>RELEASE DATE</Form.Label>
          <Form.Input
            name={"release_date"}
            placeholder={"Release Date"}
            onChange={handleChange}
            value={release_date}
            type="date"
          />
          <Form.Icon>
            <FontAwesomeIcon icon={faCalendarDays} color={"#f64261"} />
          </Form.Icon>
        </Form.RowItem>
      </Form.Row>
      <Form.Row>
        <Form.RowItem>
          <Form.Label>MOVIE URL</Form.Label>
          <Form.Input
            type="text"
            name={"image_url"}
            placeholder={"https://"}
            onChange={handleChange}
            value={image_url}
          />
        </Form.RowItem>
        <Form.RowItem>
          <Form.Label>RATING</Form.Label>
          <Form.Input
            type="text"
            name={"rating"}
            placeholder={"Rating"}
            onChange={handleChange}
            value={rating}
          />
        </Form.RowItem>
      </Form.Row>
      <Form.Row>
        <Form.RowItem>
          <Form.Label>GENRE</Form.Label>
          <DropDown
            selectedItems={genres}
            optionList={GenresList}
            onChange={handleChangeGenres}
          />
        </Form.RowItem>
        <Form.RowItem>
          <Form.Label>RUNTIME</Form.Label>
          <Form.Input
            name={"duration"}
            placeholder={"Minutes"}
            onChange={handleChange}
            value={duration}
          />
        </Form.RowItem>
      </Form.Row>
      <Form.Row>
        <Form.RowItem>
          <Form.Label>OVERVIEW</Form.Label>
          <Form.TextArea
            name="description"
            placeholder={"Movie Description"}
            onChange={handleChange}
            value={description}
          />
        </Form.RowItem>
      </Form.Row>
      <Form.Footer>
        <Form.ButtonSecondary onClick={handleReset}>RESET</Form.ButtonSecondary>
        <Form.ButtonPrimary onClick={handleSubmit}>SUBMIT</Form.ButtonPrimary>
      </Form.Footer>
    </Form.Content>
  );
};

const Delete = () => {
  return (
    <Form.Content>
      <Form.Title>DELETE MOVIE</Form.Title>
      <Form.Message>Are you sure you want to delete this movie?</Form.Message>
      <Form.Footer>
        <Form.ButtonPrimary>CONFIRM</Form.ButtonPrimary>
      </Form.Footer>
    </Form.Content>
  );
};

export default {
  Default,
  Delete,
};
