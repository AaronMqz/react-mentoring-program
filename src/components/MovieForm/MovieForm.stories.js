import MovieForm from "./MovieForm";
import { initialMovieState } from "../../utils";

export default {
  title: "Example/MovieForm",
  component: MovieForm.Default,
  tags: ["autodocs"],
  argTypes: {
    handleSubmit: { action: "Form submit" },
    handleShandleResetubmit: { action: "Form reset" },
  },
};

export const AddMovie = {
  args: {
    titleForm: "DEFAULT TITLE",
    movie: initialMovieState,
  },
};
