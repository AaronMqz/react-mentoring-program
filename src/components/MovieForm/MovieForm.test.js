import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import MovieForm from "./MovieForm";
import { dummyMovieState } from "../../utils";
import renderer from "react-test-renderer";

afterEach(cleanup);

describe("MovieForm Tests:", () => {
  test("1. That component is visible.", () => {
    render(
      <MovieForm.Default titleForm="DEFAULT TITLE" movie={dummyMovieState} />
    );
    const movieName = screen.getByPlaceholderText("Title");
    expect(movieName).toBeInTheDocument();
  });

  it("2. Snapshot renders correctly", () => {
    const tree = renderer
      .create(
        <MovieForm.Default titleForm="DEFAULT TITLE" movie={dummyMovieState} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
