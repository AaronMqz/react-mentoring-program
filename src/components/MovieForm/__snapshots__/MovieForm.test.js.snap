// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieForm Tests: 2. Snapshot renders correctly 1`] = `
<div
  className="sc-beqWaB bTkdGv"
>
  <label
    className="sc-eDDNvR gfVLRw"
  >
    DEFAULT TITLE
  </label>
  <div
    className="sc-gueYoa eZFNTp"
  >
    <div
      className="sc-dmqHEX bWJwCF"
    >
      <label
        className="sc-gLDzan htTcgQ"
      >
        TITLE
      </label>
      <input
        className="sc-jTrPJq eGHqHK"
        name="title"
        onChange={[Function]}
        placeholder="Title"
        type="text"
        value="Avengers: Inifinity War"
      />
    </div>
    <div
      className="sc-dmqHEX bWJwCF"
    >
      <label
        className="sc-gLDzan htTcgQ"
      >
        RELEASE DATE
      </label>
      <input
        className="sc-jTrPJq eGHqHK"
        name="release_date"
        onChange={[Function]}
        placeholder="Release Date"
        type="date"
      />
      <span
        className="sc-qRumB iUSyto"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-calendar-days "
          color="#f64261"
          data-icon="calendar-days"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M128 0c17.7 0 32 14.3 32 32V64H288V32c0-17.7 14.3-32 32-32s32 14.3 32 32V64h48c26.5 0 48 21.5 48 48v48H0V112C0 85.5 21.5 64 48 64H96V32c0-17.7 14.3-32 32-32zM0 192H448V464c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V192zm64 80v32c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16V272c0-8.8-7.2-16-16-16H80c-8.8 0-16 7.2-16 16zm128 0v32c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16V272c0-8.8-7.2-16-16-16H208c-8.8 0-16 7.2-16 16zm144-16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16V272c0-8.8-7.2-16-16-16H336zM64 400v32c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16V400c0-8.8-7.2-16-16-16H80c-8.8 0-16 7.2-16 16zm144-16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16V400c0-8.8-7.2-16-16-16H208zm112 16v32c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16V400c0-8.8-7.2-16-16-16H336c-8.8 0-16 7.2-16 16z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </span>
    </div>
  </div>
  <div
    className="sc-gueYoa eZFNTp"
  >
    <div
      className="sc-dmqHEX bWJwCF"
    >
      <label
        className="sc-gLDzan htTcgQ"
      >
        MOVIE URL
      </label>
      <input
        className="sc-jTrPJq eGHqHK"
        name="image_url"
        onChange={[Function]}
        placeholder="https://"
        type="text"
        value="https://m.media-amazon.com/images/I/71eHZFw+GlL._AC_SY879_.jpg"
      />
    </div>
    <div
      className="sc-dmqHEX bWJwCF"
    >
      <label
        className="sc-gLDzan htTcgQ"
      >
        RATING
      </label>
      <input
        className="sc-jTrPJq eGHqHK"
        name="rating"
        onChange={[Function]}
        placeholder="Rating"
        type="text"
        value="8.4"
      />
    </div>
  </div>
  <div
    className="sc-gueYoa eZFNTp"
  >
    <div
      className="sc-dmqHEX bWJwCF"
    >
      <label
        className="sc-gLDzan htTcgQ"
      >
        GENRE
      </label>
      <div
        className="sc-irTswW eqsVTd"
      >
        <div
          className="sc-csCMJt CoLXb"
          onClick={[Function]}
        >
          Action, Adventure
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-caret-down "
            color="#f64261"
            data-icon="caret-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="sc-dmqHEX bWJwCF"
    >
      <label
        className="sc-gLDzan htTcgQ"
      >
        RUNTIME
      </label>
      <input
        className="sc-jTrPJq eGHqHK"
        name="duration"
        onChange={[Function]}
        placeholder="Minutes"
        value="2h 29min"
      />
    </div>
  </div>
  <div
    className="sc-gueYoa eZFNTp"
  >
    <div
      className="sc-dmqHEX bWJwCF"
    >
      <label
        className="sc-gLDzan htTcgQ"
      >
        OVERVIEW
      </label>
      <textarea
        className="sc-fsQiph iYXtjn"
        name="description"
        onChange={[Function]}
        placeholder="Movie Description"
        value="As the Avengers and their allies have continued to protect the world from threats too large for any one hero to handle, a new danger has emerged from the cosmic shadows: Thanos. A despot of intergalactic infamy, his goal is to collect all six Infinity Stones, artifacts of unimaginable power, and use them to inflict his twisted will on all of reality. Everything the Avengers have fought for has led up to this moment, the fate of Earth and existence has never been more uncertain."
      />
    </div>
  </div>
  <div
    className="sc-hLseeU cTNrkv"
  >
    <div
      className="sc-jsMahE sc-kFuwaP dKEZuS fSQRFv"
    >
      RESET
    </div>
    <div
      className="sc-jsMahE dKEZuS"
    >
      SUBMIT
    </div>
  </div>
</div>
`;
